File: App/lib/logic/settings/ordered_item.dart
enum GraphKind {
  event,
  line,
  bar,
}

class OrderedItem {
  bool visible = true;
  int order = 0;
  String name;
  int id;
  GraphKind graph;

  OrderedItem(this.id, this.name, this.graph);

  OrderedItem.fromJson(dynamic json)
      : visible = json["visible"] as bool,
        id = json['id'] as int,
        order = json['order'] as int,
        name = json['name'] as String,
        graph = GraphKind.values.byName(json['graph'] as String);

  Map<String, dynamic> toJson() => {
        'visible': visible,
        'order': order,
        'name': name,
        'id': id,
        'graph': graph.name,
      };
}

File: Backend/Api/Models/Settings/OrderedItem.cs
namespace Api.Models.Settings;

public class OrderedItem
{
    public bool Visible { get; set; } = true;

    public bool ShowOnDashboard { get; set; } = true;

    public int? Order { get; set; }

    public required string Name { get; set; }

    public required long Id { get; set; }

    public string? Key { get; set; }

    public GraphKind Graph { get; set; }

    public GraphKind DetailGraph { get; set; }

    public long? Parent { get; set; }

    public long Color { get; set; }
}